src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioDocumentTrackerProvider.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal class DefaultVisualStudioDocumentTrackerProvider : VisualStudioDocumentTrackerProvider
```

Comment:
Had to split this from the factory in order to prevent a circular dependency with the parser factory.

Reply:
This is a gross hack - is this necessary because the document tracker also needs to be able to create parsers?

It might be less of a gross hack to just build **all** of the initialization into the tracker factory.

Either way you decide, document this decision in comments so it's not lost.

Reply:
Also as long as it's internal, I don't super care

Reply:
## Another Idea

Should we just have a **BufferIntializer** service? And then have all of the other things call into it? It might be more straightforward to have all of the wire-ups in one place.

Reply:
> Should we just have a BufferIntializer service? And then have all of the other things call into it? It might be more straightforward to have all of the wire-ups in one place

You thinking the service would be the bit that inserts all the tracker + the parser? Or were you thinking it'd be a one stop shop for `GetXYZFromBuffer`?